IUPAC_SMILES+.asciidoc

```diff
@@ -1125,38 +1170,36 @@ Very few SMILES systems actually implement the rules for `SP`, `TB` or `OH` chir
 == Parsing Termination
 
 A SMILES string is terminated by a whitespace terminator character (space, tab, newline,
-carriage-return), or by the end of the string.
+carriage-return), or by the end of the string. As a result, and leading space in a SMILES string
+is considered invalid in IUPAC SMILES+ (e.g. `' CCC'`). Further an empty SMILES string,`''`, is invalid.
 
 Other data or information, such as a name, properties, registration number, etc., may follow the
 SMILES on a line after the whitespace character.  SMILES parsers will ignore this data, although
 applications that use the SMILES parser will often make use of it.
 
 == Programming Practices
 
-OpenSMILES is designed to facilitate exchange of chemical information.  To achieve that goal, it
-SMILES parsers should impose as few limits as possible on the language.
-
 There is no formal limit to the length of a SMILES string; SMILES of over 1 million characters
-have been assembled for various purposes.  There is no requirement that a SMILES parser must be able
+have been assembled for various purposes. There is no requirement that a SMILES parser must be able
 to parse these exceptionally long SMILES, but as a guideline, all implementations of SMILES parsers
-should, at a minimum, accept and correctly parse SMILES strings of 100,000 characters.  If a SMILES
+should, at a minimum, accept and correctly parse SMILES strings of 100,000 characters. If a SMILES
 parser encounters a string that is too long to parse, it should generate a relevant error
 message.
 
 A SMILES parser should accept at least four digits for the <<atomclass,atom class>>,
 and the values 0 to 9999.
 
-There is no formal limit to the number of rings a molecule can contain.  There are only
-100 ring-closure numbers, but since numbers can be reused, a molecule can potentially have more
-than 100 rings.  SMILES parsers should accept and correctly parse molecules with at least 1000 rings;
+There is no formal limit to the number of rings a molecule can contain. There are only
+999 ring-closure numbers, but since numbers can be reused, a molecule can potentially have more
```

Comment:
Since '0' is a valid ring number, I believe there are 1000 ring-closure numbers, or am I wrong?